pandas/core/missing.py

```diff
@@ -645,6 +645,44 @@ def fill_zeros(result, x, y, name, fill):
     return result
 
 
+def mask_zero_div_zero(x, y, result):
+    """
+    Set results of 0 / 0 or 0 // 0 to np.nan, regardless of the dtypes
+    of the numerator or the denominator
+
+    Parameters
+    ----------
+    x : ndarray
+    y : ndarray
+    result : ndarray
+
+    Returns
+    -------
+    filled_result : ndarray
```

Comment:
the semantics you have here are not explained well. you are mutating result in-place, then returning it. is that intentional?

Reply:
I'll flesh it out.  The semantics were meant to follow fill_zeros as closely as possible.

Reply:
ok some more expl would be helpful. filling result is ok, you just have to make this really clear.